src/aleph_client/asynchronous.py

```diff
@@ -234,23 +254,49 @@ async def create_aggregate(
 
 async def create_store(
         account: Account,
-        address=settings.ADDRESS_TO_USE,
+        address: Optional[str] = None,
         file_content: Optional[bytes] = None,
+        file_path: Optional[str] = None,
         file_hash: Optional[str] = None,
-        guess_mime_type: bool = False,
+        guess_mime_type: Optional[bool] = None,
```

Comment:
IMO that's pushing the logic a bit far to have this as an optional. Any value that doesn't have a default in `settings` does not need to be optional.

Reply:
The problem with these default values is their non-transitiveness, as explained in #84. Am I getting something wrong doing it like this?

Reply:
I'm not sure I get what you mean by (non-)transitivity. I get your issue with using the defaults from the `settings` module. But here you have a boolean value which does not depend on a setting, has a sensible default value (false here) and can be overridden by the user. I don't see why changing it to an optional improves the situation.

Reply:
Hm, yeah, setting a boolean yourself is less complex than referencing to the settings, you're right

Reply:
But still, my issue with this "non-transitiveness" is the following: If at any point of functions calling functions with the same kwargs I want to change the default, I'll have to do it at the beginning of the chain, not at the end, where the value is actually required.

My issue lies with the fact that users, using the SDK, might need to write their own default handling as pointed out in the issue. This is also the case with the booleans, which creates unnecessary overhead for the user. If the functions do not care whether they are passed `None` or not and still use their respective default, then this is ideal behavior.

Reply:
It is indeed the most stupid behavior that `None` overwrites defaults.

Reply:
The cases you commented on here are the ones that do not have this chaining of function calls. But if a user decides to write their own module with the SDK and writes a function that passes its arguments to this function, then the developer will have the same headache as I had before.

It is a really weird situation that feels a lot like the `null` and `undefined` double-trouble with Javascript.

Reply:
My opinion here is that "inner" functions should never try to load a setting from a global configuration object. It introduces the whole mess you're talking about. And you fixed that, so that's good. Now, on the topic of nulls. The good practice in Python is to:

1. Use type hints. if you say something is a `bool`, then the user better pass a bool and not `None`.
2. If you want to really enforce the type hints, a good old 

```
if not isinstance(boolean_value, bool):
  raise TypeError("Get rekt")
```

is the way to go. We should not allow `None` where it doesn't make sense, period. As I mentioned earlier, the good defaults are:

```
def f_with_defaults(b: bool, b_with_sensitive_default: bool = False, l: Optional[List] = None, d: Optional[Dict] = None):
  l = l or []
  d = d or {} 
```